system-tests-new/prebuild/test_prebuild.py

```diff
@@ -73,26 +75,58 @@ def test_repeatable_fmod_hashes(self, tmp_path):
 
     def test_vanilla_prebuild(self, tmp_path):
         # share a prebuild from a different folder
-        compile_file = 'fab.steps.compile_fortran.CompileFortran.compile_file'
-        parse_fortran = 'fab.tasks.fortran.FortranAnalyser._parse_file'
 
         # build the project in "some other fab workspace", from where we'll share its prebuild
-        other_config = self.build_config(fab_workspace=tmp_path / 'other_workspace')
-        other_config.run()
+        first_project = self.build_config(fab_workspace=tmp_path / 'other_workspace')
+        first_project.run()
 
         # now build the project in our workspace.
-        my_config = self.build_config(fab_workspace=tmp_path / 'my_workspace',
-                                      prebuild_folder=other_config.prebuild_folder)
-        with mock.patch(parse_fortran) as mock_parse_fortran:
-            with mock.patch(compile_file) as mock_compile_file:
-                my_config.run()
+        second_project = self.build_config(fab_workspace=tmp_path / 'my_workspace',
+                                           grab_prebuild_folder=first_project.prebuild_folder)
+        with mock.patch('fab.tasks.fortran.FortranAnalyser._parse_file') as mock_parse_fortran:
+            with mock.patch('fab.steps.compile_fortran.CompileFortran.compile_file') as mock_compile_file:
+                second_project.run()
 
-        # make sure we didn't call the analyser
+        # make sure we didn't call the analyser or compiler
         mock_parse_fortran.assert_not_called()
-
-        # make sure we didn't call the compiler
         mock_compile_file.assert_not_called()
 
         # make sure the exe was built
-        exe = my_config.project_workspace / 'my_prog.exe'
+        exe = second_project.project_workspace / 'my_prog.exe'
         assert exe.exists()
+
+        # make sure the prebuild files are the same
+        for a in file_walk(first_project.prebuild_folder):
+            b = second_project.prebuild_folder / a.name
+            assert files_identical(a, b)
```

Comment:
This will trap files present in "a" but not in "b". Does it trap files present in "b" but not in "a"?